pkg/config/shortnames/util.go

```diff
@@ -0,0 +1,150 @@
+package shortnames
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/image/v5/pkg/sysregistriesv2"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/storage/pkg/unshare"
+	"github.com/pkg/errors"
+)
+
+//// ALGORITHMIC UTILITY ////
+
+// linear search returning location of element in a slice (returns -1 if not found)
+func find(a []string, x string) int {
+	for i, n := range a {
+		if x == n {
+			return i
+		}
+	}
+	return len(a)
+}
+
+// removeIndex removes an element at an index in a slice
+func removeIndex(s []string, index int) []string {
+	return append(s[:index], s[index+1:]...)
+}
+
+// removeDuplicates removes duplicate values in given slice
+func removeDuplicates(s []string) []string {
+	for i := 0; i < len(s); i++ {
```

Comment:
Please use a `presentStrings map[string]struct{}` (meaning a `set<string>`), and maybe build a result slice element by element from scratch instead of removing by copying, so that this can be O(N) instead of O(N^3).

(Or maybe drop it entirely if there won’t be a caller.)